pkg/chunk/chunk_store.go

```diff
@@ -151,6 +151,12 @@ func (c *Store) calculateDynamoWrites(userID string, chunks []Chunk) (WriteBatch
 
 // Get implements ChunkStore
 func (c *Store) Get(ctx context.Context, from, through model.Time, allMatchers ...*metric.LabelMatcher) ([]local.SeriesIterator, error) {
+	now := model.Now()
+	if through.After(now) {
+		util.WithContext(ctx).Warnf("Capping TimeSpanHi from future=%v to now=%v", through, now)
```

Comment:
I agree, we shouldn't log this normally, I'm open to logging at debug.
In grafana easy to accidentally click a time range that queries into the future (e.g. "this month").